src/dbus/client/thread_api_dbus.cpp

```diff
@@ -150,6 +150,7 @@ DBusHandlerResult ThreadApiDBus::DBusMessageFilter(DBusConnection *aConnection,
     {
         f(role);
     }
+    return DBUS_HANDLER_RESULT_HANDLED;
```

Comment:
I think here the filter is an observer and should not interfere with other components that look for these messages, so the return value should be `DBUS_HANDLER_RESULT_NOT_YET_HANDLED`.

Reply:
DBusMessageFilter() calls 'device role handlers' on line 151, which mean it handled the dbus message with 'device role change'? So it should return 'DBUS_HANDLER_RESULT_HANDLED'?

`for (const auto &f : mDeviceRoleHandlers)
    {
        f(role);
    }`

Reply:
I was worried that other filters may not be called, but it should be fine since there are no other filters. LGTM. Thanks.